src/map/skill.cpp

```diff
@@ -7476,6 +7476,79 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 
 		break;
 
+	case NPC_IMMUNE_PROPERTY:
+		switch (skill_lv) {
+			case 1:
+				type = SC_IMMUNE_PROPERTY_NOTHING;
+				break;
+			case 2:
+				type = SC_IMMUNE_PROPERTY_WATER;
+				break;
+			case 3:
+				type = SC_IMMUNE_PROPERTY_GROUND;
+				break;
+			case 4:
+				type = SC_IMMUNE_PROPERTY_FIRE;
+				break;
+			case 5:
+				type = SC_IMMUNE_PROPERTY_WIND;
+				break;
+			case 6:
+				type = SC_IMMUNE_PROPERTY_DARKNESS;
+				break;
+			case 7:
+				type = SC_IMMUNE_PROPERTY_SAINT;
+				break;
+			case 8:
+				type = SC_IMMUNE_PROPERTY_POISON;
+				break;
+			case 9:
+				type = SC_IMMUNE_PROPERTY_TELEKINESIS;
+				break;
+			case 10:
+				type = SC_IMMUNE_PROPERTY_UNDEAD;
+				break;
+		}
+		clif_skill_nodamage(src,bl, skill_id,skill_lv,sc_start(src,bl,type,100,skill_lv,skill_get_time(skill_id,skill_lv)));
+		sc_start(src,bl, SC_IMMUNE_PROPERTY,100,skill_lv,skill_get_time(skill_id,skill_lv));
+		break;
+
+	case NPC_MOVE_COORDINATE:
```

Comment:
I found the skill in that form, isn't that correct too?
```c
	case NPC_MOVE_COORDINATE:
		if (src != bl) {
			int x = src->x, y = src->y;
			if (unit_movepos(src,bl->x,bl->y,0,0)) {
				clif_skill_nodamage(src,src,skill_id,skill_lv,1);
				clif_blown(src);
				if (unit_movepos(bl,x,y,0,0)) {
					clif_skill_nodamage(bl,bl,skill_id,skill_lv,1);
					clif_blown(bl);
				}
				map_foreachinallrange(unit_changetarget,src,AREA_SIZE,BL_MOB,bl,src);
			}
		}
		break;

Reply:
Skill needs two kinds of form. I see this has only a single form.
Skill needs ```clif_skill_damage``` to show effect on target.
Also, ```clif_blown``` will not remove specific SCs must be removed on moving, for example SC_ROLLINGCUTTER.